arelle/ValidateXbrl.py

```diff
@@ -580,6 +608,7 @@ def checkLinks(self, modelLinks):
                                 resourceArcTos.append((toLabel, arcElt.get("use"), arcElt))
                         elif self.isGenericArc(arcElt):
                             arcrole = arcElt.get("{http://www.w3.org/1999/xlink}arcrole")
+                            assert arcrole is not None
```

Comment:
This seems like arelle needs to handle gracefully. Probably produce a schema error or something appropriate? Here and other places. Reasoning is that arelle is being used to validate docs so it _should_ handle and report any error it finds with a filing back to the user

Reply:
Agreed. @ggravlingen you don't need to handle error logging here (kind of out of scope for a type hints pr), but I'd avoid adding this assertion. Replace it with an if statement to guard against a missing arcrole:
```pytohn
if arcrole is not None:
    self.genericArcArcroles.add(arcrole)
    if arcrole in (XbrlConst.elementLabel, XbrlConst.elementReference):
        resourceArcTos.append((toLabel, arcrole, arcElt))
```

The other None assertions seem fine (they'd cause errors anyway.)

Reply:
Thanks @stevenbronson-wk, I missed that comment.

I did a little investigation and the arcrole can't be None here (Arelle blows up earlier). Ultimately 2 would be best, but I'm okay with the assertion.

```
[AttributeError] Unrecoverable error: 'NoneType' object has no attribute 'startswith', wk-20220331.htm - ../IXDS wk-20220331.htm
Traceback (most recent call last):
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 146, in load
    modelDocument = pluginMethod(modelXbrl, normalizedUri, filepath, isEntry=isEntry, namespace=namespace, **kwargs)
  File "/Users/austinmatherne/git/arelle/arelle/plugin/inlineXbrlDocumentSet.py", line 121, in inlineXbrlDocumentSetLoader
    ixdoc = load(modelXbrl, elt.text, referringElement=elt, isDiscovered=True)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 332, in load
    modelDocument.inlineXbrlDiscover(rootNode)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1326, in inlineXbrlDiscover
    self.schemaLinkbaseRefsDiscover(inlineElement)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1088, in schemaLinkbaseRefsDiscover
    self.schemaLinkbaseRefDiscover(element)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1095, in schemaLinkbaseRefDiscover
    return self.discoverHref(element, urlRewritePluginClass="ModelDocument.InstanceSchemaRefRewriter")
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1215, in discoverHref
    doc = _newDoc(self.modelXbrl, url, isDiscovered=not nonDTS, base=self.baseForElement(element), referringElement=element)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 325, in load
    modelDocument.schemaDiscover(rootNode, isIncluded, isSupplemental, namespace)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 925, in schemaDiscover
    self.schemaDiscoverChildElements(rootElement, isSupplemental)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 981, in schemaDiscoverChildElements
    self.schemaDiscoverChildElements(modelObject, isSupplemental)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 981, in schemaDiscoverChildElements
    self.schemaDiscoverChildElements(modelObject, isSupplemental)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 973, in schemaDiscoverChildElements
    self.schemaLinkbaseRefDiscover(modelObject)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1095, in schemaLinkbaseRefDiscover
    return self.discoverHref(element, urlRewritePluginClass="ModelDocument.InstanceSchemaRefRewriter")
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1215, in discoverHref
    doc = _newDoc(self.modelXbrl, url, isDiscovered=not nonDTS, base=self.baseForElement(element), referringElement=element)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 327, in load
    modelDocument.linkbaseDiscover(rootNode)
  File "/Users/austinmatherne/git/arelle/arelle/ModelDocument.py", line 1169, in linkbaseDiscover
    if XbrlConst.isDimensionArcrole(arcrole) and not dimensionArcFound:
  File "/Users/austinmatherne/git/arelle/arelle/XbrlConst.py", line 763, in isDimensionArcrole
    return arcrole.startswith("http://xbrl.org/int/dim/arcrole/")
AttributeError: 'NoneType' object has no attribute 'startswith'
```

Reply:
I take it I should leave this as is then?